src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -62,10 +64,12 @@ public void DeleteDocuments(Term term, IState state)
             _indexWriter.DeleteDocuments(term, state);
         }
 
-        public void Commit(IState state)
+        public void Commit(IState state, IndexingStatsScope commitStats)
         {
             try
             {
+                _directory.SetCommitStats(commitStats);
+                _mergeScheduler.SetCommitStats(commitStats);
```

Comment:
I think it would be better to release this after the commit, in some using so if something goes wrong we will not stay with 'leftovers'

Reply:
done